docs-src/concepts/what-is-versioning.md

```diff
@@ -0,0 +1,170 @@
+Each Compatible Version Set has a Default Build ID.
+
+A Task Queue has a default maximum of 10 Compatible Version Sets and a maximum of 100 Build IDs for a single Task Queue across all Compatible Version Sets.
+
+You can visualize Compatible Version Sets as a tree-like structure, where a branch represents the progressive evolution of each build. Each branch in the tree is a series of versions that are compatible with each other, but not necessarily with versions in other branches.
+
+For example, Build ID of `1.0` might start a new branch and therefore isn't compatible with Build ID of `0.4` from an earlier branch. However, Build ID of `1.1` is compatible with Build ID of `1.0` because it's a newer version of the same branch.
+
+## Task Queue-based Versioning
+
+While Task Queue-based Versioning is available to use, it's recommended to use the [Worker Versioning](#worker-versioning) approach instead, as it simplifies the process of versioning your Task Queues.
+
+Task Queue-based Versioning is the process of creating a new Task Queue for each version of your Workflow, and then modify the Task Queue the Client points to.
+
+Workers pull from Task Queues, and because of that you start Workflows on Task Queues, and each Task Queues have a unique identifier.
+
+Rather than creating a new Workflow Type for each version of your Workflow, you can create a new Task Queue for each version of your Workflow.
+
+This approach is similar to Workflow Type versioning, but instead of creating a new Workflow Type for each version of your Workflow, you create a new Task Queue for each version of your Workflow.
+
+This helps maintain clean libraries without branching logic.
+
+### Advantages of Task Queue-based Versioning
+
+Task Queue-based Versioning is valuable in several respects:
+
+- **Conceptually easy**: Task Queues are similar to Workflow Type versioning, but with Task Queues, so you don't need to manually update all starters to use the latest Workflow Type.
+- **Easy maintenance**: In comparison to Patching, Task Queue-based Versioning requires less maintenance because you don't need to track which Workflows are running on which version of the Workflow Type.
+
+### Disadvantages of Task Queue-based Versioning
+
+Task Queue Versioning has several disadvantages:
+
+- **Operationally complex**: To run multiple versions of a Task Queue, you need to keep old Workers alive and change which Task Queue Clients point to. This can result in a number of sets of old Workers, which operationally can be complex to manage.
+- **Bug fixes**: This approach can't be used to fix a bug in a currently running or open Workflow. Because the Workflow is already running, you are unable to change the Task Queue it's running on, and thus you can't make changes or fixes to the Workflow.
+
+## Patching
+
+Patching allows for the update of Workflow Definitions without affecting currently running Workflows.
```

Comment:
```suggestion
Patching allows for the update of Workflow Definitions without breaking currently running Workflows.
```

It'd be pretty useless if it didn't affect them somehow :smile: - what's important is it doesn't cause nondeterminism in them.